src/adaptation/Adaptation.jl

```diff
@@ -0,0 +1,40 @@
+module Adaptation
+
+import Base: string
+import LinearAlgebra, Statistics
+using ..AdvancedHMC: DEBUG
+
+abstract type AbstractAdapter end
+
+include("stepsize.jl")
+include("precond.jl")
+
+abstract type AbstractCompositeAdapter <: AbstractAdapter end
+
+# TODO: generalise this to a list of adapters
```

Comment:
Not sure if we really want a generalised composite adapter to take a list of adapters? I guess not all composition would be valid or meaningful.

Reply:
I still find the `adaption` type design hard to reason about, but I'm happy to leave this as is for now as long as it's correct.

Reply:
I guess one of the reason the current `adaption` type are hard to reason is because I introduce too many levels of abstractions. I could merge some or even make some math ones into individual modules, e.g. the variance and covariance estimaters.